src/util/log.d

```diff
@@ -350,18 +354,71 @@ class FileLogger(alias Layout) : Logger
     }
 }
 
-class RollingFileLogger(alias Layout) : FileLogger!Layout
+mixin template CachedAppender()
+{
+    import std.process : thisThreadID;
+
+    private static struct ThreadState
+    {
+        Appender!(char[]) appender;
+
+        int depth;
+    }
+
+    private ThreadState[typeof(thisThreadID)] threadStates;
+
+    private ref ThreadState threadState() return
```

Comment:
Newspaper style: put helper below the functions, which call the helper.

Reply:
class is gone